File: webdev-sp19-seung-son/src/services/UserService.js
import url from "./Source";

const SOURCE = url;

function isJson(str) {
    try {
        JSON.parse(str);
    } catch (e) {
        return false;
    }
    return true;
}

class UserService {

    register = (user) =>
        fetch(SOURCE + "/api/register", {
            method: "POST",
            headers: {'Content-Type': 'application/json'},
            credentials: 'include',
            body: JSON.stringify(user)
        }).then(response => {
            return response.text().then(function (text) {
                return isJson(text) ? JSON.parse(text) : null
            })
        });
    profile = () =>
        fetch(SOURCE + "/api/profile", {
            credentials: 'include'
        })
            .then(response => {
                return response.text().then(function (text) {
                    return isJson(text) ? JSON.parse(text) : null
                })
            });
    login = user =>
        fetch(SOURCE + '/api/login', {
            method: "POST",
            headers: {'Content-Type': 'application/json'},
            credentials: 'include',
            body: JSON.stringify(user)
        }).then(response => {
            return response.text().then(function (text) {
                return isJson(text) ? JSON.parse(text) : null
            })
        })
    logout = () =>
        fetch(SOURCE + '/api/logout', {
            method: "POST",
            credentials: 'include'
        });
    findUserById = userId =>
        fetch(SOURCE + "/api/users/" + userId)
            .then(response =>
                response.json());
    findAllUsers = () =>
        fetch(SOURCE + "/api/users")
            .then(response =>
                response.json());
    updateUser = user => {
        return fetch(SOURCE + '/api/profile', {
            method: "PUT",
            headers: {'Content-Type': 'application/json'},
            credentials: 'include',
            body: JSON.stringify(user)
        }).then(response => response)
    }
}

export default UserService;

File: whiteboard-sp19/src/main/java/edu/northeastern/cs5610/services/UserService.java
package edu.northeastern.cs5610.services;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import edu.northeastern.cs5610.models.Faculty;
import edu.northeastern.cs5610.models.Person;
import edu.northeastern.cs5610.models.Student;
import edu.northeastern.cs5610.models.Widget;
import edu.northeastern.cs5610.repositories.PersonRepository;

@RestController
@CrossOrigin(origins = "*", allowCredentials = "true", allowedHeaders = "*")
public class UserService {
	@Autowired
	PersonRepository rep;
	
	@PostMapping("/api/register")
	public Person register(@RequestBody Person user, HttpSession session) {
		List<Person> registered = (List<Person>) rep.findAll();
		for (int i = 0; i < registered.size(); i++) {
			Person p = registered.get(i);
			if (p.getUsername().equals(user.getUsername())) {
				return null;
			}
		}
		
		Faculty newUser = new Faculty(null, null, user.getUsername(), user.getPassword(), null, null, null, null);
		session.setAttribute("currentUser", newUser);
		rep.save(newUser);
		return newUser;
	}

	@GetMapping("/api/profile")
	public Person profile(HttpSession session) {
		Person currentUser = (Person)session.getAttribute("currentUser");
		return currentUser;
	}

	@PostMapping("/api/login")
	public Person login(@RequestBody Person credentials, HttpSession session) {
		List<Person> users = (List<Person>) rep.findAll();
		for (int i = 0; i < users.size(); i++) {
			Person user = users.get(i);
			if (user.getUsername().equals(credentials.getUsername())
					&& user.getPassword().equals(credentials.getPassword())) {
				session.setAttribute("currentUser", user);
				return user;
			}
		}
		return null;
	}

	@PostMapping("/api/logout")
	public void logout(HttpSession session) {
		session.invalidate();
	}

	@GetMapping("/api/users")
	public List<Person> findAllUsers() {
		return (List<Person>) rep.findAll();
	}

	@GetMapping("/api/users/{id}")
	public Person findUserById(@PathVariable("id") Integer id) {
		return rep.findById(id).get();
	}

	@PutMapping("/api/profile")
	public Person updateUser(@RequestBody Person updater, HttpSession session) {
		return rep.save(updater);
	}
}
